spring-cloud-function-adapters/spring-cloud-function-grpc/src/test/java/org/springframework/cloud/function/grpc/GrpcInteractionTests.java

```diff
@@ -300,6 +304,17 @@ public void testBiStreamStringInStreamOutFailure() {
 		}
 	}
 
+	private int patientlyGetPort(ConfigurableApplicationContext context) throws InterruptedException {
+		Thread.sleep(500);
+		String port = context.getEnvironment().getProperty("local.grpc.server.port");
+		if (port == null) {
+			Thread.sleep(500);
+			port = context.getEnvironment().getProperty("local.grpc.server.port");
+			assertThat(port).as("Unable to get 'local.grpc.server.port' - server may not have started up").isNotNull();
+		}
+		return Integer.valueOf(port);
+	}
+
```

Comment:
Are you familiar with [Awaitility](http://www.awaitility.org/)?

Reply:
Yes, I ❤️ Awaitility. It would be great here. Thx for the reminder - I will amend.

Reply:
Updated. @sbrannen do you mind giving another scan?